File: patterns/go/Intermediate/pyramid.go
package main
import (
	"fmt"
	"strings"
)

func pyramid(n int) {
	for i := 1; i <= n; i++ {
		fmt.Println(strings.Repeat(" ", n-i) + strings.Repeat("*", 2*i-1))
	}
}

func main() {
	pyramid(5)
}

File: patterns/cpp/Intermediate/pyramid.cpp
#include <iostream>
#include <string>
using namespace std;

void pyramid(int n=5){
    for(int i=1;i<=n;i++){
        cout << string(n-i, ' ') << string(2*i-1, '*') << "\n";
    }
}

int main(){
    pyramid(5);
    return 0;
}
